Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -600,7 +631,9 @@ protected VariableManaCost handleManaXCosts(Game game, boolean noMana, Player co
                             "Announce the value for " + variableManaCost.getText(), game, this);
                     int amountMana = xValue * variableManaCost.getXInstancesCount();
                     StringBuilder manaString = threadLocalBuilder.get();
-                    if (variableManaCost.getFilter() == null || variableManaCost.getFilter().isGeneric()) {
+                    if (false) { // TODO the conditional
```

Comment:
You can add additional field to `VariableManaCost` like `boolean uniqueX`. So `cost.setMaximumCost(5);` will be `cost.mustBeUnique();` with auto-setup minX/maxX inside.